### ISSUE TITLE
[Bug]: Caused by java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by ajaybirla-fabzen** [OneSignal/OneSignal-Unity-SDK#685]
### What happened?

The application crashes whenever it receives a notification.

### Steps to reproduce?

```Markdown
1. Launch the application.
2. Send a notification to the application.
3. Click on the notification.
4. Observe the application crash.
```

### What did you expect to happen?

I expect the app to not crash whenever it receives a notification.

### Unity version

2021.3.17f1

### OneSignal Unity SDK version

5.0.5

### Platform

Android

### Relevant log output

```Shell
Caused by java.util.ConcurrentModificationException:
       at java.util.ArrayList$Itr.next(ArrayList.java:860)
       at com.onesignal.common.modeling.ModelStore.persist(ModelStore.kt:140)
       at com.onesignal.common.modeling.ModelStore.addItem(ModelStore.kt:108)
       at com.onesignal.common.modeling.ModelStore.addItem$default(ModelStore.kt:98)
       at com.onesignal.common.modeling.ModelStore.add(ModelStore.kt:46)
       at com.onesignal.common.modeling.IModelStore$DefaultImpls.add$default(IModelStore.kt:37)
       at com.onesignal.core.internal.operations.impl.OperationRepo.internalEnqueue(OperationRepo.kt:76)
       at com.onesignal.core.internal.operations.impl.OperationRepo.enqueueAndWait(OperationRepo.kt:68)
       at com.onesignal.internal.OneSignalImp$login$2.invokeSuspend(OneSignalImp.kt:299)
       at com.onesignal.internal.OneSignalImp$login$2.invoke(:8)
       at com.onesignal.internal.OneSignalImp$login$2.invoke(:2)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1$1.invokeSuspend(ThreadUtils.kt:61)
       at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
       at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
       at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284)
       at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
       at kotlinx.coroutines.BuildersKt.runBlocking(:1)
       at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
       at kotlinx.coroutines.BuildersKt.runBlocking$default(:1)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:60)
       at com.onesignal.common.threading.ThreadUtilsKt$suspendifyOnThread$1.invoke(ThreadUtils.kt:59)
       at kotlin.concurrent.ThreadsKt$thread$thread$1.run(Thread.kt:30)
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jinliu9508** [comment]
@ajaybirla-fabzen, thank you for providing the log output! We've acknowledged this issue and plan to address it in the next release.

**Comment by jinliu9508** [comment]
Fixed in 5.0.6.
